src/relayclient/KnownRelaysManager.ts

```diff
@@ -91,8 +112,20 @@ export default class KnownRelaysManager implements IKnownRelaysManager {
       }
     })
     const origRelays = Array.from(activeRelays.values())
-    const filteredRelays = origRelays.filter(this.relayFilter)
-    filteredRelays.forEach(relay => this.activeRelays.add(relay))
+    return origRelays.filter(this.relayFilter)
+  }
+
+  async _fetchPreferredRelayManagers (): Promise<Set<Address>> {
```

Comment:
why "pre-ping" the preferred relays? there's no need. just accept them as valid.

Reply:
moreover, you ping them linearly: that is, if I have 5 prefered relays, then I ping 5 times in sequence, then send one "burst" of 3 pings to select the fastest. 
that means if any prefered relay is down, I'll have time TIMEOUT before continue to the next one (in the pre-ping. the normal "raceToSuccess ping" works ok.

Reply:
fixed